Sources/Sentry/SentryCoreDataTracker.m

```diff
@@ -0,0 +1,83 @@
+
+#import "SentryCoreDataTracker.h"
+#import "SentryHub+Private.h"
+#import "SentryLog.h"
+#import "SentrySDK+Private.h"
+#import "SentryScope+Private.h"
+#import "SentrySpanProtocol.h"
+
+@implementation SentryCoreDataTracker
+
+- (NSArray *)managedObjectContext:(NSManagedObjectContext *)context
+              executeFetchRequest:(NSFetchRequest *)request
+                            error:(NSError **)error
+                      originalImp:(NSArray *(NS_NOESCAPE ^)(NSFetchRequest *, NSError **))original
+{
+    __block id<SentrySpan> fetchSpan;
+    [SentrySDK.currentHub.scope useSpan:^(id<SentrySpan> _Nullable span) {
+        fetchSpan = [span startChildWithOperation:SENTRY_COREDATA_FETCH_OPERATION
+                                      description:[self descriptionFromRequest:request]];
+    }];
+    NSArray *result = original(request, error);
+
+    [fetchSpan setDataValue:[NSNumber numberWithInteger:result.count] forKey:@"result_amount"];
+
+    [fetchSpan
+        finishWithStatus:error != nil ? kSentrySpanStatusInternalError : kSentrySpanStatusOk];
+
+    return result;
+}
+
+- (BOOL)managedObjectContext:(NSManagedObjectContext *)context
+                        save:(NSError **)error
+                 originalImp:(BOOL(NS_NOESCAPE ^)(NSError **))original
+{
+
+    __block id<SentrySpan> fetchSpan = nil;
+    if (context.hasChanges) {
+        [SentrySDK.currentHub.scope useSpan:^(id<SentrySpan> _Nullable span) {
+            fetchSpan = [span startChildWithOperation:SENTRY_COREDATA_SAVE_OPERATION
+                                          description:@"Saving Database"];
+        }];
+    }
+
+    BOOL result = original(error);
+
+    [fetchSpan
+        finishWithStatus:*error != nil ? kSentrySpanStatusInternalError : kSentrySpanStatusOk];
+
+    return result;
+}
+
+- (NSString *)descriptionFromRequest:(NSFetchRequest *)request
+{
+    NSMutableString *result =
+        [[NSMutableString alloc] initWithFormat:@"FETCH '%@'", request.entityName];
```

Comment:
Even though CoreData's method is `fetch`, we're building a SQL-Like string here (with WHERE and ORDER BY, all caps) so lets go all-in with "SQL":

```suggestion
        [[NSMutableString alloc] initWithFormat:@"SELECT '%@'", request.entityName];
```